features/steps/thens/attributes.py

```diff
@@ -73,3 +73,31 @@ def _():
     misc.handle_errors(context, list(_()))
 
 
+@then('The type of attribute {attribute} must be {value}')
+def step_impl(context, attribute, value):
+    if getattr(context, 'applicable', True):
+        errors = []
+        for instance in context.instances:
+            if isinstance(instance, tuple):
+                instance = instance[0]
+            attribute_value = getattr(instance, attribute, 'Attribute not found')
+            if isinstance(attribute_value, str) or (not attribute_value[0].is_a(value)):
+                errors.append(err.AttributeTypeError(instance, attribute_value, attribute, value))
+
+        misc.handle_errors(context, errors)
+
+@then('The value of attribute {attribute} must be {value}')
+def step_impl(context, attribute, value):
+    if getattr(context, 'applicable', True):
+        errors = []
+        for instance in context.instances:
+            if isinstance(instance, tuple):
+                attribute_value = getattr(instance[0], attribute, 'Attribute not found')
+            else:
+                attribute_value = getattr(instance, attribute, 'Attribute not found')
+
+            if attribute_value != value:
```

Comment:
How does this behave with different underlying types (e.g. int, string, instance)?

Reply:
`@then('The value of attribute {attribute} must be {value}')`

`{value}` is casted to a string by behave parser. For the purposes of the rules currently developed that's ok. 
For general purpose it's more complex, but I think it's very difficult to introduce type checks without assuming the .feature file writer is familiar with types.